libs/blocks/global-footer/global-footer.js

```diff
@@ -0,0 +1,358 @@
+        el.setAttribute('daa-lh', el.previousElementSibling.textContent);
+      }
+
+      [...el.children].forEach(analyticsDecorateList);
+    }
+  };
+
+  toggleMenu = (e) => {
+    const button = e.target.closest('[role=button]');
+    const expanded = button.getAttribute('aria-expanded');
+    if (expanded === 'true') {
+      this.closeMenu(button);
+    } else {
+      this.openMenu(button);
+    }
+  };
+
+  closeMenu = (el) => {
+    if (el.id === 'region-button') {
+      window.removeEventListener('keydown', this.closeOnEscape);
+      window.removeEventListener('click', this.closeOnDocClick);
+    }
+    el.setAttribute('aria-expanded', false);
+  };
+
+  openMenu = (el) => {
+    const type = el.classList[0];
+    const expandedMenu = document.querySelector(`.${type}[aria-expanded=true]`);
+    if (expandedMenu) { this.closeMenu(expandedMenu); }
+    if (el.id === 'region-button') {
+      window.addEventListener('keydown', this.closeOnEscape);
+      window.addEventListener('click', this.closeOnDocClick);
+    }
+    el.setAttribute('aria-expanded', true);
+  };
+
+  toggleOnKey = (e) => {
+    if (e.code === 'Space' || e.code === 'Enter') {
+      this.toggleMenu(e);
+    }
+  };
+
+  setDesktopTitles = (titles) => {
+    titles?.forEach((title) => {
+      title.removeAttribute('role');
+      title.removeAttribute('tabindex');
+      title.removeAttribute('aria-expanded');
+      title.removeAttribute('aria-controls');
+
+      window.removeEventListener('keydown', this.toggleOnKey);
+      title.removeEventListener('click', this.toggleMenu);
+    });
+  };
+
+  setMobileTitles = (titles) => {
+    titles?.forEach((title) => {
+      title.setAttribute('role', 'button');
+      title.setAttribute('tabindex', 0);
+      title.setAttribute('aria-expanded', false);
+      title.setAttribute('aria-controls', `${title.id}-menu`);
+      title.addEventListener('click', this.toggleMenu);
+
+      title.addEventListener('focus', () => {
+        window.addEventListener('keydown', this.toggleOnKey);
+      });
+      title.addEventListener('blur', () => {
+        window.removeEventListener('keydown', this.toggleOnKey);
+      });
+    });
+  };
+
+  onMediaChange = (e) => {
+    const footerTitles = document.querySelectorAll('.footer-nav-item-title');
+    if (e.matches) {
+      this.setDesktopTitles(footerTitles);
+    } else {
+      this.setMobileTitles(footerTitles);
+    }
+  };
+}
+
+async function fetchFooter(url) {
+  const resp = await fetch(`${url}.plain.html`);
+  const respText = await resp.text();
+
+  if (!resp.ok) {
+    return { error: respText }; // can pass additional Response info if needed
+  }
+
+  return { html: respText };
+}
+
+export default async function init(block) {
+  const { prefix } = locale;
+  const url = getMetadata('footer-source') || `${prefix}/footer`;
+  if (url) {
+    const { html, error } = await fetchFooter(url);
+    if (error) {
+      console.log(`Could not create footer: ${error}`);
```

Comment:
Instead of logging to the `console`, have you considered using lana?

Reply:
Yep; again, the footer is identical to the one used in Milo today on production, we're re-writing it. Our todo list for MVP includes adding LANA logs wherever neccessary.

Reply:
hey @honstar, here's the production code that @overmyheadandbody  mentioned https://github.com/adobecom/milo/blob/main/libs/blocks/footer/footer.js#L344. We do have plans to use lana for logging, our version of the footer is still work in progress.